### DISCUSSION TITLE
Unable to Connect to Device in TCPIP Mode using webadb

### DISCUSSION THREAD

**Discussion by noonowl** [yume-chan/ya-webadb#604] [Q&A] [answered]
Hello,

I tried connecting to a device operating in TCPIP mode using `webadb` and ADB, but it wasn’t successful.

Upon inspecting the code, it seems that it uses `connect` from `node:net`, which appears not to function in the browser. Both `AdbDaemonDirectSocketDevice` and `TCPSocket` codes also use `connect` from `node:net`, making it impossible to establish a connection from a React browser.

Is there any other way to make the connection possible? The Android device is connected to `192.168.0.100:5555`, and I would like to establish a connection using `webadb`.

Thank you.

**Comment by yume-chan** [comment, marked as answer]
It can't. Browsers provide no API to create TCP connections.

`AdbDaemonDirectSocketDevice` uses [Direct Socket API](https://github.com/WICG/direct-sockets), which will add TCP connections to browsers, but it's still work-in-progress. `TCPSocket` type is from Direct Socket API.

**Reply by noonowl** [reply]
Thank you for your confirmation. Following what I’ve observed in another discussion, I've successfully established a connection from React to the backend via a WebSocket where `websockify` operates, and from the backend to ADB via TCP/IP. I agree that having DirectSocket support would be beneficial, but for the time being, utilizing WebSocket seems to be the most effective solution available.
